metadata-ingestion/src/datahub/ingestion/source/sql/snowflake.py

```diff
@@ -539,17 +539,38 @@ def do_provision_role_internal(self):
                 [
                     f"grant usage on DATABASE {db_name} to role {role}",
                     f"grant usage on all schemas in database {db_name} to role {role}",
-                    f"grant select on all tables in database {db_name} to role {role}",
-                    f"grant select on all external tables in database {db_name} to role {role}",
-                    f"grant select on all views in database {db_name} to role {role}",
                     f"grant usage on future schemas in database {db_name} to role {role}",
-                    f"grant select on future tables in database {db_name} to role {role}",
                 ]
             )
+            if self.config.profiling.enabled:
```

Comment:
@anshbansal, are there any mutually conflicting privileges between the two lists from in the `if` and `else`? Just want to make sure as that this switch can be turned on/off arbitrarily without causing issues with being able to grant the roles (i.e. no privilege needs to be revoked before granting these).